README.md
kiwi-calls
==========

Application for displaying kiwi calls


## Development setup

For the best results use nginx to serve the media files during development. To do this on ubuntu 12.04

- install the latest nginx, add the following to `/etc/apt/sources.list.d/nginx.list`
```  
    deb http://nginx.org/packages/ubuntu/ precise nginx
    deb-src http://nginx.org/packages/ubuntu/ precise nginx
```

- run
```
    wget -qO - http://nginx.org/keys/nginx_signing.key | sudo apt-key add -
    apt-get update
    apt-get install nginx
```    

- something similar to /etc/nginx/conf.d/default.conf
```
    server {
        listen   80;  # or say 9000 if you want to run apache
        server_name foo.dragonfly.co.nz;
        # no security problem here, since / is alway passed to upstream
        # serve directly - analogous for static/staticfiles
	    location /media/sonograms/ {
	        alias /kiwi/sonograms/;
	        expires 30d;
	    }
	    location /media/snippets/ {
	        alias /kiwi/snippets/;
	        expires 30d;
	    }
        location /media/ {
    	    alias /home/foo/dragonfly/songscape/media/;
    	    expires 30d;
        }
        location /static/ {
    	    alias /home/foo/dragonfly/songscape/static/;
    	    expires 30d;
        }
        location / {
            proxy_pass_header Server;
            proxy_set_header Host $http_host;
            proxy_redirect off;
            proxy_set_header X-Real-IP $remote_addr;
            proxy_set_header X-Scheme $scheme;
            proxy_connect_timeout 10;
            proxy_read_timeout 10;
            proxy_pass http://localhost:8000/;
        }
        # what to serve if upstream is not available or crashes
        error_page 500 502 503 504 /media/50x.html;
    }
```


debian/changelog
songscape (20140124-1) precise; urgency=low

  * Remove the collect static line to get the postinst to run

 -- Edward Abraham <edward@victoria.dragonfly.co.nz>  Fri, 24 Jan 2014 19:06:24 +1300

songscape (20140122-4) precise; urgency=low

  * Remove the --upgrade flag on pip

 -- Edward Abraham <edward@dragonfly.co.nz>  Wed, 22 Jan 2014 21:22:04 +1200

songscape (20140118-4) precise; urgency=low

  * Upgrade wasnt triggered on server

 -- Edward Abraham <edward@dragonfly.co.nz>  Sat, 18 Jan 2014 21:22:04 +1200

songscape (20140118-3) precise; urgency=low

  * OK. Local settings seem to be copied to the wrong place

 -- Edward Abraham <edward@dragonfly.co.nz>  Sat, 18 Jan 2014 20:22:04 +1200

songscape (20140118-2) precise; urgency=low

  * Bump to update local settings

 -- Edward Abraham <edward@dragonfly.co.nz>  Sat, 18 Jan 2014 20:22:04 +1200

songscape (20140118-1) precise; urgency=low

  * Change the site URL to https

 -- Edward Abraham <edward@dragonfly.co.nz>  Sat, 18 Jan 2014 19:22:04 +1200

songscape (20140110-3) precise; urgency=low

  * Modify the postinst

 -- Edward Abraham <edward@dragonfly.co.nz>  Fri, 10 Jan 2014 19:22:04 +1200

songscape (20140110-2) precise; urgency=low

  * Specify matplotlib

 -- Edward Abraham <edward@dragonfly.co.nz>  Fri, 10 Jan 2014 18:22:04 +1200

songscape (20140110-1) precise; urgency=low

  * Change the dependencies

 -- Edward Abraham <edward@dragonfly.co.nz>  Fri, 10 Jan 2014 17:22:04 +1200

songscape (20130508-1) precise; urgency=low

  * Initial version

 -- Joel Pitt <joel@haast.dragonfly.co.nz>  Wed, 08 May 2013 17:22:04 +1200


www/settings.py
# Django settings for www project.
import os

DEBUG = True
TEMPLATE_DEBUG = DEBUG

ADMINS = (
    ('Edward Abraham', 'edward@dragonfly.co.nz'),
)

# This directory
PROJECT_DIR = os.path.dirname(__file__)

MANAGERS = ADMINS

DATABASES = {
    'default': {
        'ENGINE': 'django.db.backends.postgresql_psycopg2',
        'NAME': 'songscape_dev',
        'USER': 'dba',
        'PASSWORD': '',
        'HOST': 'localhost',
        'PORT': '',
    }
}

TIME_ZONE = 'UTC'
LANGUAGE_CODE = 'en-nz'
USE_I18N = True
USE_L10N = True
USE_TZ = True

SITE_ID = 1

# Absolute filesystem path to the directory that will hold user-uploaded files.
MEDIA_ROOT = os.path.join(PROJECT_DIR, 'media')

# URL that handles the media served from MEDIA_ROOT. Make sure to use a
# trailing slash.
# Examples: "http://media.lawrence.com/media/", "http://example.com/media/"
MEDIA_URL = '/media/'

STATIC_ROOT = os.path.join(PROJECT_DIR, '../static/')

STATICFILES_DIRS = (
    # Don't forget to use absolute paths, not relative paths.
    os.path.join(PROJECT_DIR, 'static'),
)

TEMPLATE_DIRS = (
    # Put strings here, like "/home/html/django_templates" or "C:/www/django/templates".
    # Always use forward slashes, even on Windows.
    # Don't forget to use absolute paths, not relative paths.
    os.path.join(PROJECT_DIR, "templates"),
)

# URL prefix for static files.
# Example: "http://media.lawrence.com/static/"
STATIC_URL = '/static/'

ADMIN_MEDIA_PREFIX = '/static/admin/'

# List of finder classes that know how to find static files in
# various locations.
STATICFILES_FINDERS = (
    'django.contrib.staticfiles.finders.FileSystemFinder',
    'django.contrib.staticfiles.finders.AppDirectoriesFinder',
)


# List of callables that know how to import templates from various sources.
TEMPLATE_LOADERS = (
    'django.template.loaders.filesystem.Loader',
    'django.template.loaders.app_directories.Loader',
#     'django.template.loaders.eggs.Loader',
)

TEMPLATE_CONTEXT_PROCESSORS = (
   'django.contrib.auth.context_processors.auth',
   'django.core.context_processors.request',
   'django.core.context_processors.media',
   'django.core.context_processors.debug',
   'django.contrib.messages.context_processors.messages',
   'django.core.context_processors.request',
)

MIDDLEWARE_CLASSES = (
    'django.middleware.common.CommonMiddleware',
    'django.contrib.sessions.middleware.SessionMiddleware',
    'django.middleware.csrf.CsrfViewMiddleware',
    'django.contrib.auth.middleware.AuthenticationMiddleware',
    'django.contrib.messages.middleware.MessageMiddleware',
#    'debug_toolbar.middleware.DebugToolbarMiddleware',
)

ROOT_URLCONF = 'www.urls'

LOGIN_REDIRECT_URL = '/'

INSTALLED_APPS = (
    'django.contrib.auth',
    'django.contrib.contenttypes',
    'django.contrib.sessions',
     'django.contrib.sites',
    'django.contrib.messages',
    'django.contrib.staticfiles',
    'django.contrib.admin',
    'django.contrib.admindocs',
    'www.recordings',
    'south',
    'django_nose',
    #'debug_toolbar',
)

TEST_RUNNER = 'django_nose.NoseTestSuiteRunner'
NOSE_ARGS = [ '--with-xunit', '--with-doctest', ]

INTERNAL_IPS = ('127.0.0.1',)

# A sample logging configuration. The only tangible logging
# performed by this configuration is to send an email to
# the site admins on every HTTP 500 error.
# See http://docs.djangoproject.com/en/dev/topics/logging for
# more details on how to customize your logging configuration.
LOGGING = {
    'version': 1,
    'disable_existing_loggers': False,
    'filters': {
    'require_debug_false': {
            '()': 'django.utils.log.RequireDebugFalse',
        }
    },
    'handlers': {
        'mail_admins': {
            'level': 'ERROR',
            'filters': ['require_debug_false'],
            'class': 'django.utils.log.AdminEmailHandler',
        },
        'console':{
            'level': 'DEBUG',
            'class': 'logging.StreamHandler',
        },
    },
    'loggers': {
        'django.request': {
            'handlers': ['mail_admins'],
            'level': 'ERROR',
            'propagate': True,
            'filters': ['require_debug_false'],
        },
        'django_browserid': {
            'handlers': ['console'],
            'level': 'DEBUG',
        },
    }
}

RECORDINGS_PATH = '' # Path of the raw recordings

# organisation repositories. A dictionary with keys
# being the organisation codes, and values
# being the URLs of the servers that hold the
# raw recordings
#REPOSITORIES = {'RFPT': 'http://rfpt.songscape.org'}
REPOSITORIES = {'RFPT': 'http://192.168.0.123:8888'}

# Set your site url for security
SITE_URL = 'http://localhost:8000'

BROWSERID_AUDIENCES = ['http://localhost:8000',]

DEBUG_TOOLBAR_CONFIG = {'INTERCEPT_REDIRECT': False}

# Add the django_browserid authentication backend.
AUTHENTICATION_BACKENDS = (
   'django.contrib.auth.backends.ModelBackend', # required for admin
)


# Path to redirect to on successful login.
LOGIN_REDIRECT_URL = '/'

# Path to redirect to on unsuccessful login attempt.
LOGIN_REDIRECT_URL_FAILURE = '/'

# Path to redirect to on logout.
LOGOUT_REDIRECT_URL = '/'

CACHES = {
    'default': {
        'BACKEND': 'django.core.cache.backends.locmem.LocMemCache',
        'LOCATION': 'unique-snowflake'
    }
}



# Audio and sonogram quality settings
MAX_FRAMERATE = 8000
MIN_FREQ=500
MAX_FREQ=4000
N_FFT=512


# MEDIA_ROOT can be overridden in local_settings
SONOGRAM_DIR = 'sonograms/'
SNIPPET_DIR = 'snippets/'

try:
    from local_settings import *
except ImportError:
    print "Error loading local settings"
    pass


import sys
#if manage.py test was called, use test settings
if 'test' in sys.argv or 'migrationcheck' in sys.argv:
    try:
        from .test_settings import *
    except ImportError:
        print "Can't find test_settings.py"

